database/neo4j/neo4j.go

```diff
@@ -256,7 +261,20 @@ func (n *Neo4j) ensureVersionConstraint() (err error) {
 		}
 	}()
 
-	query := fmt.Sprintf("CREATE CONSTRAINT ON (a:%s) ASSERT a.version IS UNIQUE", n.config.MigrationsLabel)
+	// Get constraint and check to avoid error duplicate
+	res, err := neo4j.Collect(session.Run("CALL db.constraints", nil))
+	if err != nil {
+		return err
+	}
+
+	for _, record := range res {
```

Comment:
I'm not familiar with neo4j. Is there a better way to ensure that the index/constraint is only created once? This may be quite slow for DBs with many constraints.

Also, what happens if 2 processes attempt to create the constraint at the same time? It's common to automatically run `migrate` when deploying an application onto multiple servers.

Are there other docs around constraints than [this](https://neo4j.com/docs/cypher-manual/current/administration/constraints/)?

Reply:
Hi @dhui 
I updated the code, the new code only returns a single result

```
Also, what happens if 2 processes attempt to create the constraint at the same time? It's common to automatically run migrate when deploying an application onto multiple servers.
```
I am also new with neo4j, I cannot answer this, neo4j has MERGE but cannot be used to create constraint